File: Strings/Q2. Longest Common Prefix/longest_common_prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        strs.sort()
        
        first = strs[0]
        last = strs[-1]
        ans = ""
        
        for i in range(len(first)):
            if first[i] == last[i]:
                ans += first[i]
            else:
                break
        
        return ans
        

File: Strings/Q2. Longest Common Prefix/longestcommonprefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String first=strs[0];
        String last=strs[strs.length-1];
        StringBuilder st=new StringBuilder();
        
        for(int i =0 ;i<Math.min(first.length(),last.length());i++){
            char ch1=first.charAt(i);
            char ch2=last.charAt(i);
            if(ch1==ch2) st.append(ch1);
            else if(ch1!=ch2) break;
            
        }
    return st.toString();
    }
}